From 165958248e123bbff8f91b35f12e354267ac6d58 Mon Sep 17 00:00:00 2001
From: Anton Arapov <anton@redhat.com>
Date: Thu, 15 Jul 2010 12:51:51 +0200
Subject: [PATCH] setup_helper: make r/w ops of security/limits.d/*.conf more robust.

    previous routine led to failures on
      \n, whitespace, comment
    in configuration file.

Reported-by: CAI Qian <caiqian@redhat.com>
Signed-off-by: Anton Arapov <anton@redhat.com>
---
 huge_page_setup_helper.py |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/huge_page_setup_helper.py b/huge_page_setup_helper.py
index 9de0739..c4b3b16 100755
--- a/huge_page_setup_helper.py
+++ b/huge_page_setup_helper.py
@@ -296,8 +296,13 @@ if debug == False:
     for line in limitsConfLines:
         cfgExist = False
         for hugeUser in hugePageUserList:
-            if line.split()[0] == hugeUser:
-                cfgExist = True
+            try:
+                if line.split()[0] == hugeUser:
+                    cfgExist = True
+            except IndexError:
+                # hit either white or comment line, it is safe not to take
+                # any action and continue.
+                pass
         if cfgExist == True:
             continue
         else:
-- 
1.7.1

